docs/user-manual/framework/memory-management/buffer-pool.md

```diff
@@ -263,20 +215,52 @@ Buffers bins should be tailored based on expected usage. If many small requests
 The above trivial example allows for a few small allocations and one large allocation. In this case, there is a risk that the large allocation is used for the small allocation use case and thus care should be taken to ensure that the smaller use
 cases have a sufficient number of buffers to prevent stealing of larger allocations.
 
-## Separation of `Fw::Buffer` Allocation and Deallocation
+## Deallocation strategy
 
-There is no requirement that the allocating component and the deallocating component are the same. Thus components may
-be chained together for multiple processing steps before deallocation. There are two requirements:
+`Fw::Buffer`s **must** eventually be returned to the same instance that allocated them.
 
-1. `Fw::Buffers` must eventually be returned to the instance that allocated them
-2. `Svc.StaticMemory` cannot be used in chains involving asynchronous calls
+Additionally, it is a recommended practice to deallocate a buffer in the same component where it was allocated. This is accomplished using the buffer ownership "Return-To-Sender" pattern described below.
 
-Inter-component connections typically use the same `Fw.BufferSend` port to pass the buffer along the chain. A sample
-chain is shown here using Svc.BufferManager as the allocation source.
+### Design Pattern: Return-To-Sender
 
-```fpp
-      comp1.allocate -> my_buffer_manager.bufferGetCallee
-      comp1.sendToProcess -> comp2.process
-      comp2.sendToProcessMore -> comp3.processMore
-      comp3.deallocate -> my_buffer_manager.bufferSendIn
-```
\ No newline at end of file
+In situations where a component allocates a buffer but cannot deallocate it directly (for example, when the buffer is sent asynchronously to another component), a common pattern is the buffer "return-to-sender" design pattern.
+
+Each component returns the received buffer to its immediate sender, creating an "unwrapping" effect where buffers flow forward through the processing chain and return backward step-by-step until reaching the original allocator.
+
+```mermaid
+flowchart LR
+    BM[Buffer Manager]
+    A[Component A]
+    B[Component B]
+    C[Component C]
+    
+    %% Force layout order with invisible edge
+    BM ~~~ A
+    
+    A -->|"(1) allocate"| BM
+    A -->|"(2) forward buffer"| B
+    B -->|"(3) forward buffer"| C
+    C -->|"(4) return to sender"| B
+    B -->|"(5) return to sender"| A
+    A -->|"(6) deallocate"| BM
+    
+    style BM fill:#e1f5ff
+    style A fill:#fff4e1
+    style B fill:#f0f0f0
+    style C fill:#f0f0f0
+```
+
+**Flow explanation:**
```

Comment:
If we need this much explaination of the diagram, something seems off.

Reply:
The diagram is clear on its own, I was merely adding more "explanation", but agreed that it is redundant. Will remove.